src/_pytest/assertion/rewrite.py

```diff
@@ -949,11 +949,19 @@ def visit_BinOp(self, binop):
         res = self.assign(ast.BinOp(left_expr, binop.op, right_expr))
         return res, explanation
 
+    def _is_any_call_with_generator_or_list_comprehension(self, call):
+        """Return True if the Call node is an 'any' call with a generator or list comprehension"""
+        return (
+            isinstance(call.func, ast.Name)
+            and call.func.id == "all"
+            and isinstance(call.args[0], (ast.GeneratorExp, ast.ListComp))
```

Comment:
this still needs a length check -- can you add this as a test?

```python
def test():
    f = ((1,))
    assert all(*f)
```

Reply:
Done! 👍